### PR TITLE
[mri_violations] fix back url

### PR DESCRIPTION
**Pull request by kongtiaowang**
This pull request fixes back URL from mri_violations to imaging_browser.
https://redmine.cbrain.mcgill.ca/issues/13473

### CODE REVIEW

File: modules/mri_violations/php/NDB_Menu_Filter_mri_protocol_check_violations.class.inc [hunk 1]
```diff
@@ -47,8 +47,14 @@ class NDB_Menu_Filter_Mri_Protocol_Check_Violations extends NDB_Menu_Filter
...
     function _setupVariables()
     {
-        $this->query    = " FROM mri_violations_log l LEFT JOIN mri_scan_type m
+        $UID = $_GET["SeriesUID"];
+        if ($UID == null) {
+            $this->query = " FROM mri_violations_log l LEFT JOIN mri_scan_type m
                          ON m.ID=l.Scan_type WHERE 1=1 ";
+        } else {
+            $this->query = " FROM mri_violations_log l LEFT JOIN mri_scan_type m
+                         ON m.ID=l.Scan_type WHERE l.SeriesUID = '$UID' ";
```
Comment (davidblader) [inline modules/mri_violations/php/NDB_Menu_Filter_mri_protocol_check_violations.class.inc]: SQL injection vulnerability?

File: modules/mri_violations/php/NDB_Menu_Filter_mri_protocol_check_violations.class.inc [hunk 2]
```diff
@@ -47,8 +47,20 @@ class NDB_Menu_Filter_Mri_Protocol_Check_Violations extends NDB_Menu_Filter
      */
     function _setupVariables()
     {
-        $this->query    = " FROM mri_violations_log l LEFT JOIN mri_scan_type m
+        $UID = $_GET["SeriesUID"];
+        //preventing SQL injection -- UID only takes number and dot
+        $pattern = "/^\d[0-9\.]+/";
+        $uid     = array();
```
Comment (ridz1208) [inline modules/mri_violations/php/NDB_Menu_Filter_mri_protocol_check_violations.class.inc]: @driusan do you forsee any issues with having 2 different variables with the same name and different casing ?? maybe PHP 8 ????

**Comment by MounaSafiHarab** [comment]
@kongtiaowang 

@davidblader already passed manual test on this one, but I too tried it, and I can go back now using the back URL.

File: modules/mri_violations/php/NDB_Menu_Filter_mri_protocol_check_violations.class.inc (hunk 2 above)
Comment (ridz1208) [reply modules/mri_violations/php/NDB_Menu_Filter_mri_protocol_check_violations.class.inc]: Dave does not think its an issue

File: modules/mri_violations/php/NDB_Menu_Filter_mri_protocol_check_violations.class.inc [hunk 3]
```diff
@@ -47,8 +47,20 @@ class NDB_Menu_Filter_Mri_Protocol_Check_Violations extends NDB_Menu_Filter
...
+        $match   = preg_match($pattern, $UID, $uid);
+
+        if ($match == 1) {
+
+            $this->query = " FROM mri_violations_log l LEFT JOIN mri_scan_type m
+                         ON m.ID=l.Scan_type WHERE l.SeriesUID = '$uid[0]' ";
```
Comment (driusan) [inline modules/mri_violations/php/NDB_Menu_Filter_mri_protocol_check_violations.class.inc]: This should use $DB->quote to quote $uid[0] portion to ensure that it's properly quoted.

File: modules/mri_violations/php/NDB_Menu_Filter_mri_protocol_check_violations.class.inc [hunk 4]
```diff
@@ -47,8 +47,20 @@ class NDB_Menu_Filter_Mri_Protocol_Check_Violations extends NDB_Menu_Filter
      */
     function _setupVariables()
     {
-        $this->query    = " FROM mri_violations_log l LEFT JOIN mri_scan_type m
+        $UID = $_GET["SeriesUID"];
+        //preventing SQL injection -- UID only takes number and dot
+        $pattern = "/^\d[0-9\.]+/";
```
Comment (driusan) [inline modules/mri_violations/php/NDB_Menu_Filter_mri_protocol_check_violations.class.inc]: This isn't required after adding the $DB->quote call (and if it is, this is only used one place, so you can just use the constant directly in the function call..)

**Comment by kongtiaowang** [comment]
The back url works without this bug-fix now, So I will close this PR.

**Pull request closed without merging**